src/workspace-data/WorkspaceAttributes.js

```diff
@@ -220,6 +237,64 @@ export const WorkspaceAttributes = ({
                   noContentMessage: _.isEmpty(initialAttributes) ? 'No Workspace Data defined' : 'No matching data',
                   border: false,
                   columns: [
+                    {
+                      size: { basis: 70, grow: 0 },
+                      headerRenderer: () => {
+                        return h(Fragment, [
+                          h(Checkbox, {
+                            checked: allVisibleAttributesSelected,
+                            disabled: initialAttributes.length === 0 || editIndex !== undefined,
+                            onChange: () => {},
+                            'aria-label': 'Select all',
+                          }),
+                          h(
+                            MenuTrigger,
+                            {
+                              closeOnClick: true,
+                              content: h(Fragment, [
+                                h(
+                                  MenuButton,
+                                  {
+                                    onClick: () => setSelection(Object.fromEntries(initialAttributes.map(([id]) => [id, true]))),
+                                  },
+                                  [`All (${initialAttributes.length})`]
+                                ),
+                                filteredAttributes.length < initialAttributes.length &&
+                                  h(
+                                    MenuButton,
+                                    {
+                                      onClick: () => setSelection(Object.fromEntries(filteredAttributes.map(([id]) => [id, true]))),
+                                    },
+                                    [`Filtered (${filteredAttributes.length})`]
+                                  ),
+                                h(
+                                  MenuButton,
+                                  {
+                                    onClick: () => setSelection({}),
+                                  },
+                                  ['None']
+                                ),
+                              ]),
+                              side: 'bottom',
+                            },
+                            [h(Clickable, { 'aria-label': '"Select All" options' }, [h(Icon, { icon: 'caretDown' })])]
+                          ),
+                        ]);
+                      },
```

Comment:
I find it awkward that clicking the checkbox does nothing (you have to select one of the menu items). This isn't the case with data tables - could the checkbox itself be clickable?

Reply:
Oops, forgot to add back the `onClick` handler for the checkbox.

Reply:
Updated in https://github.com/DataBiosphere/terra-ui/pull/4927/commits/2025f9e7ccd5bf389778f3c034b5e5c41beb076e.